repose-aggregator/components/filters/api-validator-filter/src/main/java/org/openrepose/filters/apivalidator/ApiValidatorHandler.java

```diff
@@ -153,9 +146,16 @@ public void doFilter(HttpServletRequest request, HttpServletResponse response, F
                 }
 
                 if (!isValid) {
-                    if (mbcsInvalidRequests != null) {
+                    if (Optional.ofNullable(metricsService).isPresent()) {
+                        MetricRegistry metricRegistry = metricsService.getRegistry();
                         for (String s : matchedRoles) {
-                            mbcsInvalidRequests.mark(s);
+                            metricRegistry.meter(
```

Comment:
Maybe, but I'm still glad you got rid of those classes. ;)

Really, those classes were just saving us from reconstructing the prefix for the metric name (i.e., the non-dynamic bit). All of the thread-safety is still here. In Scala, I think I can even give you the savings for the metric name. Let me play with that...